cache/common.go

```diff
@@ -2,6 +2,44 @@ package cache
 
 import (
 	"errors"
+	"fmt"
+
+	"github.com/buildpack/imgutil/remote"
+
+	"github.com/buildpack/lifecycle"
+	"github.com/buildpack/lifecycle/auth"
+	"github.com/buildpack/lifecycle/cmd"
```

Comment:
I know `exporter.go` currently makes this mistake, but I would prefer if nothing outside of `/cmd` imported the `cmd` package. Everything in `cmd` should be about the user interface we are providing with the CLIs. The `cmd` package probably shouldn't be required by packages that implement the core functionality.

Reply:
Inlined MaybeCache in both restorer/main and analyzer/main and added a new NewImageCacheFromName function to cache/image_cache. Does that work?